Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -1340,27 +1339,36 @@ void CompareWorkspaces::recordMismatch(const std::string &msg, std::string ws1,
   m_result = false;
 }
 
+//------------------------------------------------------------------------------------------------
+/** Function which calculates absolute error between two values and analyses if
+this error is within the limits requested.
+
+@param x1    -- first value to check difference
+@param x2    -- second value to check difference
+@param atol  -- the tolerance of the comparison. Must be nonnegative
+
+@returns true if absolute difference is within the tolerance; false otherwise
+*/
+bool CompareWorkspaces::withinAbsoluteTolerance(double const x1, double const x2, double const atol) {
+  // NOTE !(|x1-x2| > atol) is not the same as |x1-x2| <= atol
```

Comment:
very good comment

Reply:
Thanks.  For anyone from the future coming to this PR and wondering about this comment.  Mathematically they are identical, but they do not evaluate the same in some of the tests.

The checks throughout the algo were originally written to expect `false` for a match.  It was originally checking $\vert x_1 - x_2 \vert > atol$.

It seemed to make more sense to return `true` for a match.  I changed it to return $\vert x_1 - x_2 \vert \leq atol$.  This caused certain tests to fail.  I changed it to its current form, and the same tests do not fail.

Two possibilities are values such as `NaN`, or machine precision issues, which could cause the two comparison to be mechanically different.

Reply:
I was curious why. `std::nan` is equal to everything, but isn't greater-than anything